File: iosApp/iosApp/Views/Components/Title.swift
//
//  TitleModifier.swift
//  iosApp
//
//  Created by Jan Cortiel on 06.02.23.
//  Copyright © 2023 Ludwig Boltzmann Institute for
//  Digital Health and Prevention - A research institute
//  of the Ludwig Boltzmann Gesellschaft,
//  Oesterreichische Vereinigung zur Foerderung
//  der wissenschaftlichen Forschung 
//  Licensed under the Apache 2.0 license with Commons Clause 
//  (see https://www.apache.org/licenses/LICENSE-2.0 and
//  https://commonsclause.com/).
//

import SwiftUI

struct Title: View {
    var titleText: String
    var textAlignment: TextAlignment = .leading
    var body: some View {
        Text(titleText)
            .font(.more.title)
            .foregroundColor(.more.primary)
            .fontWeight(.more.title)
            .multilineTextAlignment(textAlignment)
    }
}

struct Title_Preview: PreviewProvider {
    static var previews: some View {
        Title(titleText: "Hello World")
    }
}

File: androidApp/src/main/java/io/redlink/more/app/android/shared_composables/Title.kt
/*
 * Copyright LBI-DHP and/or licensed to LBI-DHP under one or more
 * contributor license agreements (LBI-DHP: Ludwig Boltzmann Institute
 * for Digital Health and Prevention -- A research institute of the
 * Ludwig Boltzmann Gesellschaft, Österreichische Vereinigung zur
 * Förderung der wissenschaftlichen Forschung).
 * Licensed under the Apache 2.0 license with Commons Clause
 * (see https://www.apache.org/licenses/LICENSE-2.0 and
 * https://commonsclause.com/).
 */
package io.redlink.more.app.android.shared_composables

import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import io.redlink.more.app.android.ui.theme.MoreColors

@Composable
fun Title(text: String,
          modifier: Modifier = Modifier,
          color: Color = MoreColors.PrimaryDark,
          textAlign: TextAlign = TextAlign.Start,
          maxLines: Int  = 2
) {
    Text(
        text = text,
        fontWeight = FontWeight.Bold,
        fontSize = 20.sp,
        color = color,
        textAlign = textAlign,
        maxLines = maxLines,
        modifier = modifier.fillMaxWidth()
    )
}